misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,302 @@ use strict;
+  my $dbi          = $ref_arg->{dbi} // $self->dbi;
+  my $verbose      = $ref_arg->{verbose} // 0;
 
+  # initial param validation step
   if((!defined $source_id) or (!defined $species_id) or (!defined $file) ){
     croak "Need to pass source_id, species_id and file as pairs";
   }
-  $verbose |=0;
-
-  my $peptide_source_id = $self->get_source_id_for_source_name('RefSeq_peptide', 'otherfeatures', $dbi);
-  my $mrna_source_id = $self->get_source_id_for_source_name('RefSeq_mRNA', 'otherfeatures', $dbi);
-  my $ncrna_source_id = $self->get_source_id_for_source_name('RefSeq_ncRNA', 'otherfeatures', $dbi);
-
-  my $pred_peptide_source_id =
-    $self->get_source_id_for_source_name('RefSeq_peptide_predicted', 'otherfeatures', $dbi);
-  my $pred_mrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_mRNA_predicted','otherfeatures', $dbi);
-  my $pred_ncrna_source_id =
-    $self->get_source_id_for_source_name('RefSeq_ncRNA_predicted', 'otherfeatures', $dbi);
-
-  if($verbose){
-    print "RefSeq_peptide source ID = $peptide_source_id\n";
-    print "RefSeq_mRNA source ID = $mrna_source_id\n";
-    print "RefSeq_ncRNA source ID = $ncrna_source_id\n";
-    print "RefSeq_peptide_predicted source ID = $pred_peptide_source_id\n";
-    print "RefSeq_mRNA_predicted source ID = $pred_mrna_source_id\n" ;
-    print "RefSeq_ncRNA_predicted source ID = $pred_ncrna_source_id\n" ;
-  }
 
-  my $user = "ensro";
-  my $host;
-  my $port = 3306;
-  my $dbname;
-  my $pass;
-  my $transcript_score_threshold = 0.75;
-  my $tl_transcript_score_threshold = 0.75;
-  my $project;
-
-# Grep the project name, should be ensembl or ensemblgenomes
-  if($file =~ /project[=][>](\S+?)[,]/){
-    $project = $1;
-  }
+  my $file_params = $self->parse_file_string($file);
 
-# If specified, get core database connection details
-  if($file =~ /host[=][>](\S+?)[,]/){
-    $host = $1;
-  }
-  if($file =~ /port[=][>](\S+?)[,]/){
-    $port =  $1;
-  }
-  if($file =~ /dbname[=][>](\S+?)[,]/){
-    $dbname = $1;
+  # project or db param validation
+  if ( $file_params->{project} ne ('ensembl' || 'ensemblgenomes') && !defined $db ) {
+    croak "Missing or unsupported project value (supported values: ensembl, ensemblgenomes), or missing db value.";
   }
-  if($file =~ /pass[=][>](\S+?)[,]/){
-    $pass = $1;
-  }
-  if($file =~ /user[=][>](\S+?)[,]/){
-    $user = $1;
-  }
-
-  my $ofuser = 'ensro';
-  my $ofhost;
-  my $ofport = 3306;
-  my $ofdbname;
-  my $ofpass;
 
-# If specified, get otherfeatures database connection details
-  if($file =~ /ofhost[=][>](\S+?)[,]/){
-    $ofhost = $1;
-  }
-  if($file =~ /ofuser[=][>](\S+?)[,]/){
-    $ofuser = $1;
-  }
-  if($file =~ /ofport[=][>](\S+?)[,]/){
-    $ofport =  $1;
+  # set default values
+  $file_params->{user}   //= 'ensro';
+  $file_params->{port}   //= '3306';
+  $file_params->{ofuser} //= 'ensro';
+  $file_params->{ofport} //= '3306';
+
+  # set the list of source names
+  my $source_names = {
```

Comment:
This could be made read-only, just in case (it's my C++ background showing again... const, const, const).